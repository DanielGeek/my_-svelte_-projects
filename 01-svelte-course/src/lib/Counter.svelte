<script>
	export let initialCount = 0;
	export let maxCount = undefined;

	console.log($$props, $$restProps);

	// 9. Reactive Statements
	// $: string = `Count2 is ${count2}`;

	let count = initialCount;

	function increment() {
		if (count === maxCount) return;
		count += 1;
	}
</script>

<button on:click={increment} data-testid="counter">Clicks: {count} </button>

<style>
	button {
		padding: 15px 20px;
	}
</style>
