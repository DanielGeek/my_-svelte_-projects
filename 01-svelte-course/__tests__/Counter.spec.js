import { render } from '@testing-library/svelte';
import Counter from '../src/lib/Counter.svelte';

describe('Counter', () => {
	it('Snapshot', () => {
		const { component } = render(Counter);
		expect(component).toMatchSnapshot();
	});
	it('renders with default values', () => {
		const { getByTestId } = render(Counter);
		const button = getByTestId('counter');
		expect(button).toHaveTextContent('Clicks: 0');
	});

	it('increments count when button is clicked', async () => {
		const { getByTestId, container } = render(Counter);
		const button = getByTestId('counter');
		await button.click();
		expect(button).toHaveTextContent('Clicks: 1');
		expect(container.querySelector('h3')).toHaveTextContent(
			'You have clicked 1 times.'
		);
	});
});
