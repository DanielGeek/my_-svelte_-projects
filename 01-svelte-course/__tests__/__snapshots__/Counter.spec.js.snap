// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot 1`] = `
Counter {
  "$$": {
    "after_update": [],
    "before_update": [],
    "bound": {},
    "callbacks": {},
    "context": Map {},
    "ctx": [
      0,
      "You have clicked 0 times.",
      [Function],
    ],
    "dirty": [
      -1,
    ],
    "fragment": {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": [
      [Function],
    ],
    "on_disconnect": [],
    "on_mount": [],
    "props": {},
    "root": <div>
      <button
        class="svelte-17m46ol"
        data-testid="counter"
      >
        Clicks: 
        0
      </button>
       
      <h3>
        You have clicked 0 times.
      </h3>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
